tests/component/specs/viewToolbar.cy.js

```diff
@@ -224,4 +224,93 @@ describe('View Toolbar Component', () => {
     // TODO: visual regression test
     // https://github.com/cylc/cylc-ui/issues/178
   })
+
+  it('filters task states', () => {
+    mountToolbar([
+      {
+        title: 'Group 1',
+        controls: [
+          {
+            title: 'Filter',
+            action: 'taskStateFilter',
+            value: [],
+            key: 'filter'
+          }
+        ]
+      },
+    ])
```

Comment:
In recent changes I have fixed a reactivity issue (which means that the `eager` is no longer needed).

Unfortunately, that has broken this test. In order for this to work, we need to implement a `setOption` callback like we do in each view, i.e:

```vue
<ViewToolbar @setOption="setOption" />
```

I think the solution should be something along the lines of (see `listeners` in the Info View tests):

```js
import { reactive } from 'vue'

const model = reactive([])
mountToolbar({
    props: { groups: [...] },
    listeners: { setOption: (key, value) => { model = value } }
})
```

However, I've so far been unable to get the `setOption` callback to fire (even though we know the event has been emitted by inspecting the wrapper).